### PR TITLE
document-portal: add runtime (/usr) path support

### PR DESCRIPTION
**Pull request by xjuan**
Backported and finished https://github.com/cosimoc/flatpak/tree/wip/doc-portal-runtime

### CODE REVIEW

File: document-portal/xdp-main.c [hunk 1]
```diff
@@ -47,6 +47,7 @@ static GQueue get_mount_point_invocations = G_QUEUE_INIT;
 static XdpDbusDocuments *dbus_api;
 FlatpakDir *user_dir = NULL;
 FlatpakDir *system_dir = NULL;
+GHashTable *app_runtimes = NULL;
```
Comment (xjuan) [inline document-portal/xdp-main.c]: This is not really needed but I do not like the idea of parsing the metadata key file each time we need to resolve a path.
BTW is there a better way to get the runtime path from an app id?

File: document-portal/xdp-main.c [hunk 2]
```diff
@@ -383,44 +384,54 @@ static char *
...
+        flatpak_find_deploy_for_ref (runtime_ref, NULL, NULL);
+      if (!runtime_deploy)
+        return NULL;
+
+      g_autoptr(GFile) deploy_dir = flatpak_deploy_get_files (runtime_deploy);
+      if (!deploy_dir)
+        return NULL;
+
+      g_hash_table_insert (app_runtimes, g_strdup (app_id),
+                           (char *) flatpak_file_get_path_cached (deploy_dir));
```
Comment (xjuan) [inline document-portal/xdp-main.c]: I am assuming the return value here will not change during runtime

File: document-portal/xdp-main.c [hunk 3]
```diff
@@ -491,28 +490,35 @@ validate_fd (int fd,
...
         {
-          char *rel_path = path_buffer + strlen ("/newroot/app/");
-          g_autofree char *app_path = NULL;
+          g_autofree char *real_path = NULL;
 
-          app_path = resolve_flatpak_app_path (rel_path, app_id);
-          if (app_path == NULL)
+          real_path = (has_app_prefix) ?
+                       resolve_flatpak_app_path (path + strlen ("/app/"), app_id) :
+                       resolve_flatpak_usr_path (path + strlen ("/usr"), app_id);
```
Comment (cosimoc) [inline document-portal/xdp-main.c]: Only one of these has the trailing slash

**Comment by alexlarsson** [comment]
Sorry for the silence here, but I just pushed a completely different way to solve this, which also incidentally drops the requirement on systemd --user (and thus makes --no-desktop deprecated) and allows all sorts of cool things.

**Comment by xjuan** [comment]
No problem, nice to hear you fixed the issue properly :D

**Pull request closed without merging**